credential/schema/vcjsonschema_test.go

```diff
@@ -254,6 +254,53 @@ func TestIsCredentialValidForJSONSchema_JsonSchemaCredential(t *testing.T) {
 		})
 	})
 
+	t.Run("2.2 The value of the credentialSchema property MUST always be set to [known json schema]", func(t *testing.T) {
+		t.Run("valid credentialSchema", func(t *testing.T) {
+			cred := getTestVCJSONSchemaCredential()
+			schema := getTestVCJSONSchemaSchema()
+			err := IsCredentialValidForJSONSchema(cred, schema, JSONSchemaCredentialType)
+			assert.NoError(t, err)
+		})
+
+		t.Run("credentialSchema wrong id", func(t *testing.T) {
+			cred := getTestVCJSONSchemaCredential()
+			schema := getTestVCJSONSchemaSchema()
+			schema["credentialSchema"] = map[string]any{
+				"id":        "bad",
+				"type":      JSONSchemaType,
+				"digestSRI": JSONSchemaCredentialDigestSRI,
+			}
+			err := IsCredentialValidForJSONSchema(cred, schema, JSONSchemaCredentialType)
+			assert.Error(t, err)
+			assert.ErrorContains(t, err, "credential schema's credential schema id<bad> does not match known id")
+		})
+
+		t.Run("credentialSchema wrong type", func(t *testing.T) {
+			cred := getTestVCJSONSchemaCredential()
+			schema := getTestVCJSONSchemaSchema()
+			schema["credentialSchema"] = map[string]any{
+				"id":        JSONSchemaCredentialSchemaID,
+				"type":      "NotJsonSchema",
+				"digestSRI": JSONSchemaCredentialDigestSRI,
+			}
+			err := IsCredentialValidForJSONSchema(cred, schema, JSONSchemaCredentialType)
+			assert.Error(t, err)
+			assert.ErrorContains(t, err, "credential schema's credential schema type<NotJsonSchema> does not match known type<JsonSchema>")
+		})
+
+		t.Run("credentialSchema missing digestSRI", func(t *testing.T) {
```

Comment:
Can you add a test where the `digestSRI`, when computed from the source, doesn't match the one that was given in the VC?

Reply:
yes